src/App.tsx

```diff
@@ -1,61 +1,149 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+
+export type User = {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+};
+
+export type Todo = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user?: User | null;
+};
+
+const getUserById = (userId: number): User | null => {
+  return (
+    usersFromServer.find(userFromServer => userId === userFromServer.id) || null
+  );
+};
+
+export const todos: Todo[] = todosFromServer.map(todo => {
+  return { ...todo, user: getUserById(todo.userId) };
+});
 
 export const App = () => {
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState('0');
```

Comment:
The `selectedUserId` state is initialized as a string ('0'), but it is used as a number in the `newTodo` object (line 75). Consider initializing `selectedUserId` as a number (0) and updating the logic accordingly to avoid type inconsistencies.